text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+The attestation reports include information on what chaincode is running; what specific TEE is in use, including version number and hash of the chaincode (MRENCLAVE), and including the public keys of the enclave. 
+Upon creation each chaincode enclave generates public/private key pairs for signing and for encryption. The public signing key also denotes the chaincode enclave identity.
+
+In order to support the endorsement model of Fabric, chaincode enclaves executing the same FPC Chaincode need access to a set of shared secrets. In particular, these shared or common keys are used to encrypt and decrypt the transaction arguments and state.
+That is, there is also a public/private key pair and a symmetric state encryption key per FPC Chaincode that are shared among all chaincode enclaves that run the same FPC Chaincode using the key distribution protocol.
+-->
+
+A detailed description (internal view) of the FPC deployment process is provided in the FPC 1.0 Architecture section below and can be found in the [Full Detail Diagrams](#full-detail-diagrams) section.
+
+
+# Threat & Trust Model
+<!-- note: put threat-model here as threat-model is crucial to understand when looking at architecture and the application classes covered, all of this only covered here.  
+So putting it before User Experience would put it a bit in wrong context ...
+Arguably, it might even be better to put it _after_ the architecture as there are essentially already some forward references to architecture features involved.
+-->
+
+- Organizations do not have to trust each other (i.e., users, admins and software such as peers of another organization or arbitrary collusions thereof) as far as confidentiality is concerned.
+An organization can modify any software (including any hypervisor, operating system, or Fabric itself).
+Yet, such an organization would not be able to extract private state from the chaincode,
+or learn anything about the requests or responses of the victim organization (other than what the chaincode logic allows them to learn about it).
+
+- Organizations do have to trust a quorum of other organizations as defined by the lifecycle and chaincode endorsement policies as far as integrity of chaincode metadata is concerned. E.g., they have to rely on quorums of admins to only create and modify chaincodes as appropriate and they have to rely on quorum of peers to properly execute enclave registry and validation chaincode transactions.
+
+- We also assume that normally users trust the peers of their own organization, e.g., when retrieving chaincode encryption keys. (This is primarily for simplicity. As for any Fabric chaincode, users outside of organizations could implement queries without trust in a single organization/peer by repeating queries withe multiple peers/organizations until enough identical responses are received to satisfy the endorsement policy, similar to transaction validation at peers before applying them to the ledger.)
```

Comment:
I'm not following this bullet.

Reply:
This is for completeness. As described below in the Architecture Section we provide the Enclave Registry (implemented as a normal chaincode) to store “verified” FPC chaincode metadata, such as enclave identities, public keys, etc ..., on the ledger. With the assumption made in this bullet we say that a user would trust the peer of its own Org to retrieve “verified” data stored on the ledger (e.g. all the fpc metadata stored in the registry). But we also say that this assumption is just for simplicity since a user could also contact the peer of another Org and check that all responses match. I updated this bullet.